src/pages/index.js

```diff
@@ -18,42 +19,201 @@ import {
-  ".cards__list "
-);
-section.renderItems();
+});
+
+let section;
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    section = new Section(
+      {
+        items: cards,
+        renderer: (cardData) => {
+          const sectionCard = createCard(cardData);
+          section.addItem(sectionCard);
+        },
+      },
+      ".cards__list "
+    );
+
+    section.renderItems();
+  })
+  .catch((error) => {
+    console.error("Error fetching initial cards", error);
+  });
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
 // User Info
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const userInfo = new UserInfo({
+  nameSelector: ".profile__title",
+  descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
+
+// Function to handle click event
+function handleImageClick({ name, link }) {
+  popupWithImage.open(name, link);
+}
+
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLike
+  );
+  return card.getView();
+}
+
+function handleProfileEditSubmit({ name, description }) {
+  profileEditPopup.setLoading(true);
+  api
+    .updateUserInfo(name, description)
+    .then(() => {
+      userInfo.setUserInfo({ name: name, description: description });
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditPopup.setLoading(false);
+      profileEditPopup.close();
+    });
+}
+
+function handleAddImageSubmit(name, url) {
+  addCardPopup.setLoading(true);
+  api
+    .addCard(name, url)
+    .then((cardData) => {
+      const card = createCard(cardData);
+      section.addItem(card);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addCardPopup.setLoading(false);
+    });
+}
+
+function handleAvatarSubmit(url) {
+  profileAvatarPopup.setLoading(true);
+  api
+    .updateAvatar(url)
+    .then((userData) => {
+      userInfo.setAvatar(userData);
+      profileAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileAvatarPopup.setLoading(false);
+    });
+}
+
+function handleDeleteCard(cardId) {
+  cardDeletePopup.open();
+  cardDeletePopup.setSubmitAction(() => {
+    cardDeletePopup.setLoading(true, "Deleting...");
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        this.handleDeleteCard();
+        cardDeletePopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        cardDeletePopup.setLoading(false, "Yes");
+      });
+  });
+}
+
+function handleLike(cardId) {
+  if (cardId._isLiked) {
+    api
+      .dislikeCard(cardId._id)
```

Comment:
`_isLiked`
`_id`


Please, delete the **low dash** (`underscore`) from methods if you want to use them `outside` a class because the `low dash` shows a **private** method which is used only inside the class.